openmc/deplete/coupled_operator.py

```diff
@@ -416,7 +417,81 @@ def _generate_materials_xml(self):
         for mat in self.materials:
             mat._nuclides.sort(key=lambda x: nuclides.index(x[0]))
 
+        # Remove decay-only nuclides
+        #mat_ids = [mat.id for mat in self.materials]
+        material_decay_nuclides = self._remove_decay_nuclides(self.burnable_mats)
+
         self.materials.export_to_xml()
+        if bool(material_decay_nuclides):
+            self._add_decay_nuclides(self.burnable_mats, material_decay_nuclides)
+
+    def _remove_decay_nuclides(
+        self, material_ids: List[str]
+    ) -> Dict[str, List[Tuple[str, float, str]]]:
```

Comment:
So I'm thinking an easier way to handle this would be to pass a list of nuclides to `self.materials.export_to_xml()` that should be ignored at the point of writing the XML files. That way we don't need to remove them and add them back in, and the functionality also could hypothetically be used outside of depletion. What do you think?

Reply:
Yeah that seem like a more robust solution to me.

Reply:
I will implement this tomorrow.